### DISCUSSION TITLE
Is there a way to create two templates in one file?

### DISCUSSION THREAD

**Discussion by rajvatsal** [stevearc/overseer.nvim#358] [Q&A] [answered]
I have `C++SingleFile_debug.lua` & `C++SingleFile_release.lua` templates. I've tried to do this but it doesn't work. Error is something like `template.table` not found
```lua
return {
   { --[[ debug template ]] },
   { --[[ release template ]] },
}
```
I want something like sublime text where you can add variants to a build script which will differentiate between release and debug versions. Creating them in separate files is tedious.

**Comment by DavidePatria** [comment]
did you find a way to do so? I came here looking for something like this

**Comment by stevearc** [comment, marked as answer]
I think you're looking for [template providers](https://github.com/stevearc/overseer.nvim/blob/master/doc/guides.md#template-providers)
